src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,55 @@
+package duke.task;
+
+import duke.exception.DukeException;
+import duke.exception.DukeIllegalFormatException;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Deadline class encapsulates a deadline task.
+ */
+public class Deadline extends Task {
+    /**
+     * Deadline
+     */
+    protected LocalDate by;
+    /**
+     * Deadline Label ie. "D"
+     */
+    private static final String LABEL = "D";
+
+    /**
+     * Constructs a Deadline with the specified description and deadline.
+     *
+     * @param description Description.
+     * @param by          Deadline.
+     * @throws DukeException If either the description or deadline does not follow the format.
+     */
+    public Deadline(String description, String by) throws DukeException {
+        super(description);
+        if (by.isEmpty()) {
+            throw new DukeIllegalFormatException("☹ OOPS!!! The deadline cannot be empty.");
+        }
+        try {
+            this.by = LocalDate.parse(by);
+        } catch (DateTimeParseException e) {
+            throw new DukeIllegalFormatException(
+                    "☹ OOPS!!! Seems like you have entered a wrong date format. " +
+                            "Try this instead: YYYY-MM-DD"
```

Comment:
Should the indentation of this line be the same as line 39 as well? I believe it is ok for all wrapped lines to be at the same indentation level!

Reply:
Hmm, I have been letting the formatter do the work haha. Will take note of it nex time!